src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioDiagnosticListTable.cs

```diff
@@ -1,49 +1,57 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System.ComponentModel;
-using System.ComponentModel.Composition;
 using System.Linq;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
 using Microsoft.VisualStudio.LanguageServices.Implementation.TaskList;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
 using Microsoft.VisualStudio.Shell.TableManager;
-using Microsoft.VisualStudio.TaskStatusCenter;
+using Task = System.Threading.Tasks.Task;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.TableDataSource
 {
-    using Workspace = Microsoft.CodeAnalysis.Workspace;
-
-    [Export(typeof(VisualStudioDiagnosticListTable))]
     internal partial class VisualStudioDiagnosticListTable : VisualStudioBaseDiagnosticListTable
     {
         internal const string IdentifierString = nameof(VisualStudioDiagnosticListTable);
 
-        private readonly IErrorList _errorList;
         private readonly LiveTableDataSource _liveTableSource;
         private readonly BuildTableDataSource _buildTableSource;
 
-        [ImportingConstructor]
-        public VisualStudioDiagnosticListTable(
-            SVsServiceProvider serviceProvider,
-            VisualStudioWorkspace workspace,
+        private IErrorList _errorList;
+
+        public static async Task RegisterAsync(
+            Shell.IAsyncServiceProvider asyncServiceProvider,
+            VisualStudioWorkspaceImpl workspace,
+            IDiagnosticService diagnosticService,
+            ITableManagerProvider provider)
+        {
+            var table = new VisualStudioDiagnosticListTable(workspace, diagnosticService, workspace.ExternalErrorDiagnosticUpdateSource, provider);
+
+            table.SetErrorList((IErrorList)await asyncServiceProvider.GetServiceAsync(typeof(SVsErrorList)).ConfigureAwait(false));
```

Comment:
Do we need a async listener operation surrounding this so integration tests know when they can safely assume the error list is updated?